core/templates/pages/exploration-player-page/services/exploration-engine.service.spec.ts

```diff
@@ -0,0 +1,909 @@
+          },
+          labelled_as_correct: true,
+          param_changes: [],
+          refresher_exploration_id: null,
+          missing_prerequisite_skill_id: null
+        }), 1, 0, 'default_outcome');
+
+      let lastCard = stateCardObjectFactory.createNewCard(
+        'Card 1', 'Content html', 'Interaction text', null,
+        null, null, 'content_id');
+
+      spyOn(contextService, 'isInExplorationEditorPage').and.returnValue(false);
+      spyOn(playerTranscriptService, 'getLastStateName')
+        .and.returnValue('Start');
+      spyOn(playerTranscriptService, 'getLastCard').and.returnValue(lastCard);
+      spyOn(answerClassificationService, 'getMatchingClassificationResult')
+        .and.returnValue(answerClassificationResult);
+      let alertsServiceSpy = spyOn(
+        alertsService, 'addWarning').and.returnValue();
+      spyOn(learnerParamsService, 'getAllParams').and.returnValue({});
+      spyOn(explorationEngineService, 'makeParams')
+        .and.returnValue(null);
+
+      explorationEngineService.init(
+        explorationDict, 1, null, true, ['en'], initSuccessCb);
+
+      explorationEngineService.submitAnswer(
+        answer, textInputService, submitAnswerSuccessCb);
+
+      expect(alertsServiceSpy)
+        .toHaveBeenCalledWith('Parameters should not be empty.');
+    });
+
+    it('should show warning message if the question ' +
+      'name is empty', () => {
+      let initSuccessCb = jasmine.createSpy('success');
+      let submitAnswerSuccessCb = jasmine.createSpy('success');
+      let answer = 'answer';
+      let answerClassificationResult = new AnswerClassificationResult(
+        outcomeObjectFactory.createFromBackendDict({
+          dest: 'Mid',
+          feedback: {
+            content_id: 'feedback_1',
+            html: 'feedback'
+          },
+          labelled_as_correct: true,
+          param_changes: [],
+          refresher_exploration_id: null,
+          missing_prerequisite_skill_id: null
+        }), 1, 0, 'default_outcome');
+
+      let lastCard = stateCardObjectFactory.createNewCard(
+        'Card 1', 'Content html', 'Interaction text', null,
+        null, null, 'content_id');
+
+      spyOn(contextService, 'isInExplorationEditorPage').and.returnValue(false);
+      spyOn(playerTranscriptService, 'getLastStateName')
+        .and.returnValue('Start');
+      spyOn(playerTranscriptService, 'getLastCard').and.returnValue(lastCard);
+      spyOn(answerClassificationService, 'getMatchingClassificationResult')
+        .and.returnValue(answerClassificationResult);
+      spyOn(explorationEngineService, 'makeQuestion')
+        .and.returnValue(null);
+      let alertsServiceSpy = spyOn(
+        alertsService, 'addWarning').and.returnValue();
+
+      explorationEngineService.init(
+        explorationDict, 1, null, true, ['en'], initSuccessCb);
+
+      explorationEngineService.submitAnswer(
+        answer, textInputService, submitAnswerSuccessCb);
```

Comment:
I'm not sure, but I don't think there should be a new line between these function calls, since they constitute the action part of the test. The same comment for other places also.  However, I'll leave this to your discretion.

Reply:
noted!